packages/cardano-services/test/TxSubmit/TxSubmitHttpServer.test.ts

```diff
@@ -139,28 +156,52 @@ describe('TxSubmitHttpServer', () => {
 
   describe('healthy but failing submission', () => {
     describe('/submit', () => {
-      // eslint-disable-next-line max-len
-      it('returns a 400 coded response with detail in the body to a transaction containing a domain violation', async () => {
-        const stubErrors = [new Cardano.TxSubmissionErrors.BadInputsError({ badInputs: [] })];
+      const stubErrors = [new Cardano.TxSubmissionErrors.BadInputsError({ badInputs: [] })];
+
+      beforeAll(async () => {
         txSubmitProvider = {
           healthCheck: jest.fn(() => Promise.resolve({ ok: true })),
           submitTx: jest.fn(() => Promise.reject(stubErrors))
         };
         txSubmitHttpServer = TxSubmitHttpServer.create({ txSubmitProvider }, config);
         await txSubmitHttpServer.initialize();
         await txSubmitHttpServer.start();
+      });
+
+      afterAll(async () => {
+        await txSubmitHttpServer.shutdown();
+      });
+
+      it('rehydrates errors when used with TxSubmitHttpProvider', async () => {
```

Comment:
Agree, it is cleaner. But still it wont be test e2e, data mappers and req validators wont be handled

Reply:
Hmm could you please provide an example of a data mapper and a req validator? Not sure I understand what you mean, it does make a request to an actual TxSubmitHttpServer

Reply:
Agree @mkazlauskas

Reply:
Isnt the request done to the provider? Looking at your test example: https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/cardano-services/test/TxSubmit/TxSubmitHttpServer.test.ts#L179
`clientProvider` here has `TxSubmitProvider` type

Reply:
I think I know where the misunderstanding is: `TxSubmitProvider` is an interface, which is:
- a dependency of `TxSubmitHttpServer` so that we can separate the http concerns vs actual functionality
- implemented by the object that is returned from `txSubmitHttpProvider(apiUrlBase)`. The implementation of this provider makes http requests to `TxSubmitHttpServer`.